rclpy/test/test_clock.py

```diff
@@ -319,6 +378,28 @@ def run():
     assert retval is False
 
 
+def test_sleep_for_context_shut_down(non_default_context):
+    clock = Clock()
+    retval = None
+
+    def run():
+        nonlocal retval
+        retval = clock.sleep_for(Duration(seconds=999999), context=non_default_context)
+
+    t = threading.Thread(target=run, daemon=True)
+    t.start()
+
+    # wait for thread to get inside sleep_for call
```

Comment:
same

Reply:
does is matter if `shutdown` was called before or after the thread got inside `sleep_for`?
It seems that it doesn't, in that case I would remove the sleep

Reply:
It does. This test checks that the sleep method will sleep on an initially valid context and then wake if it was shutdown. If `shutdown` was called before the thread got inside sleep_for then the method would raise `NotInitializedException` instead.

Reply:
I see, I don't have a suggestion of how to improve this one.
I'm not sure if the inconsistency makes sense, as while trying to shutdown asynchronously it would trigger different behaviors depending if the wait_until()/wait_for() called already happened or not.

I think this approach is acceptable for the moment.